tests/test_pipeline.py

```
@@ -0,0 +1,136 @@
+        AbstractContextManager[StubMapping],
+    ],
+) -> None:
+    """Assert all major phases execute in the legacy ordered sequence."""
+    updater = PlacesUpdater(mock_hass, mock_config_entry, sensor)
+    call_order: list[str] = []
+
+    updater.sensor.set_attr(ATTR_LAST_PLACE_NAME, "Last Place")
+
+    now = datetime(2024, 1, 1, 12, 0, tzinfo=UTC)
+
+    def record_prev_last_place_name(key: str, default: object = None) -> str:
```

Ruff (0.15.12)

[warning] Unused function argument: `default`

(ARG001)